Snowflake.Data/Client/SnowflakeDbCommand.cs

```diff
@@ -274,6 +281,179 @@ protected override async Task<DbDataReader> ExecuteDbDataReaderAsync(CommandBeha
             }
         }
 
+        /// <summary>
+        /// Execute a query in async mode.

```

Comment:
I added a short description of what async mode does in the summary. 

For the readme, I can write a similar thing and also give sample codes of how to run queries in async mode. I'm not sure if I can update the readme in this PR because usually we would ping someone from the docs team and then they would update the readme

Reply:
Ok, let's not change README.md for now.